bentoml/yatai/client/bento_repository_api.py

```diff
@@ -200,21 +267,38 @@ def upload_from_dir(self, saved_bento_path, labels=None):
             )
 
         if response.uri.type == BentoUri.LOCAL:
-            if os.path.exists(response.uri.uri):
-                # due to copytree dst must not already exist
-                shutil.rmtree(response.uri.uri)
-            shutil.copytree(saved_bento_path, response.uri.uri)
-
-            self._update_bento_upload_progress(bento_service_metadata)
+            # For remote yatai service, call upload method.
+            if isinstance(self.yatai_service, YataiStub):
+                try:
+                    self.upload_bento(
+                        bento_service_metadata.name,
+                        bento_service_metadata.version,
+                        saved_bento_path,
+                    )
+                    upload_status = UploadStatus.DONE
+                except Exception as e:  # pylint: disable=broad-except
+                    logger.error(e)
+                    upload_status = UploadStatus.ERROR
+            else:
+                if os.path.exists(response.uri.uri):
```

Comment:
not related to this PR - this should probably raise an exception here instead, let's revisit this

Reply:
@yubozhao could you make this change in this PR and add a test?

Reply:
could you address this as well?